packages/coinjoin/src/client/round/selectRound.ts

```diff
@@ -0,0 +1,230 @@
+        .flatMap(round => {
+            const current = currentRounds.find(r => r.id === round.id);
+            if (current) return current;
+            // try to create new CoinjoinRound
+            try {
+                return generator(round, options);
+            } catch (e) {
+                // constructor fails on invalid round data (highly unlikely)
+                return [];
+            }
+        });
+};
+
+// Basic preselect Accounts
+// exclude Account.utxos which are registered in CoinjoinRounds or detained in CoinjoinPrison
+export const getAccountCandidates = (
+    accounts: Account[],
+    _statusRounds: Round[],
+    currentRounds: CoinjoinRound[],
+    prison: CoinjoinPrison,
+    { log }: CoinjoinRoundOptions,
+) => {
+    // TODO: walk thru all Round[] and search in round events for account input/output scriptPubKey which are not supposed to be there (interrupted round)
+    // if they are in phase 0 put them to prison to cool off so they registration on coordinator will timeout naturally, otherwise prison for longer, they will be banned
+
+    // collect outpoints of all currently registered inputs in CoinjoinRounds + inputs in prison
+    const registeredOutpoints = currentRounds
+        .flatMap(round => round.inputs.concat(round.failed).map(u => u.outpoint))
+        .concat(prison.inmates.map(i => i.id));
+
+    return accounts.flatMap(account => {
+        // skip account registered to critical rounds
+        const { accountKey } = account;
+        const isAccountAlreadyRegistered = currentRounds.find(
+            round =>
+                round.phase !== RoundPhase.Ended &&
+                round.inputs.find(input => input.accountKey === accountKey),
+        );
+
+        if (isAccountAlreadyRegistered) {
+            log(`Skipping candidate <${accountKey}>. Already registered to round`);
+            return [];
+        }
+
+        // TODO: double-check account max signed rounds, should be done by suite tho
+
+        // exclude account utxos which are unavailable
+        const utxos = account.utxos.filter(utxo => !registeredOutpoints.includes(utxo.outpoint));
+        if (utxos.length > 0) {
+            if (account.skipRounds) {
+                const [low, high] = account.skipRounds;
+                // skip reached lower limit
+                // or skip randomly (20% chance with [4, 5] settings)
+                if (account.skipRoundCounter >= low || Math.random() > low / high) {
+                    account.skipRoundCounter = 0;
+                    log(`Random skip candidate <${accountKey}>`);
+                    return [];
+                }
+                account.skipRoundCounter++;
+            }
+
+            log(`Found account candidate <${accountKey}> with ${utxos.length} inputs`);
+            return {
+                ...account,
+                utxos,
+            };
+        }
+
+        log(`Skipping candidate <${accountKey}>. Utxos ${utxos.length} of ${account.utxos.length}`);
```

Comment:
`utxos.length` will always be === 0 here, right? Is it intentional?

Reply:
utxos.length yes but account.utxos.length doesnt have to

utxos are picked from account.utxos so this message will tell us how many was not picked (0 of 3)